swagger.json

```diff
@@ -2494,6 +3082,70 @@
                     }
                 }
             }
+        },
+        "driverIdParam": {
+            "name": "driver_id",
+            "in": "path",
+            "required": true,
+            "description": "The driver ID.",
+            "type": "integer",
+            "format": "int64"
+        },
+        "vehicleIdParam": {
+            "name": "vehicle_id",
+            "in": "path",
+            "required": true,
+            "description": "The vehicle ID.",
+            "type": "integer",
+            "format": "int64"
+        },
+        "routeIdParam": {
+            "name": "route_id",
+            "in": "path",
+            "required": true,
+            "description": "The route ID.",
+            "type": "integer",
+            "format": "int64"
+        },
+        "routeCreateParam": {
+            "name": "createDispatchRouteParams",
+            "required": true,
+            "in": "body",
+            "schema": {
+                "$ref": "#/definitions/DispatchRouteCreate"
+            }
+        },
+        "routeUpdateParam": {
+            "name": "updateDispatchRouteParams",
+            "required": true,
+            "in": "body",
+            "schema": {
+                "$ref": "#/definitions/DispatchRoute"
```

Comment:
I think we should have a separate schema definition for `DispatchRouteUpdate` since it doesn't support setting `job_state`, `arrived_at_ms`, `completed_at_ms`, `skipped_at_ms`.

Reply:
When we ask consumers to PUT a route, we should expect them to send us exactly what we sent them in a GET request, with some values modified. I think we should still expect them to populate completed_at_ms even if they can't modify the value.

Reply:
I think Go JSON marshaler would ignore additional fields, so it's ok for users to send us extra fields noted above. 
However I think it would be confusing to the users of the API if we let the users specifying fields like `job_state` but they wouldn't have any effect.
Also, if someone manually construct an API request, I personally think it's nicer to not requiring them to specify extra fields that are functionally not necessary.